src/trip-repo.js

```diff
@@ -3,51 +3,95 @@ import Trip from "./trip.js";
 class TripRepo {
   constructor(tripData, destinationData) {
     this.allTrips = tripData;
-    this.destinationData = destinationData;;
+    this.destinationData = destinationData;
   }
 
   compareDates(d1, d2) {
-    const date1 = new Date(d1)
+    const date1 = new Date(d1);
     const date2 = new Date(d2);
-
     return date1 > date2 ? true : false;
   }
 
+  matchDestinationNames(tripArray) {
+    const linkDestination = tripArray.map(trip => {
+      const findDestination = this.destinationData.find(dest => dest.id === trip.destinationID);
+      const newTrip = new Trip(trip, findDestination);
+      return newTrip;
+    });
+    return linkDestination;
+  }
+
+  createOnlyDateDestination(tripArray, userID) {
+    const currentUserTrips = tripArray.filter(trip => trip.userID === userID);
+    const destinationArray = this.matchDestinationNames(currentUserTrips);
+    const newObject = destinationArray.map(trip => ({'date':trip.date, 'destination':trip.destination.destination}));
```

Comment:
This isn't a blocker, but try to avoid "array" and "object" when naming things as it doesn't provide context as to what the data actually represents. Awesome automatic return here though! 🎉